UpsertUserDataProcessing/handler.ts

```diff
@@ -81,54 +84,63 @@ export function UpsertUserDataProcessingHandler(
-    );
-    if (isLeft(errorOrUpsertedUserDataProcessing)) {
-      const { body } = errorOrUpsertedUserDataProcessing.value;
+    const errorOrComputedStatus = maybeRetrievedUserDataProcessing.fold<
+      Either<IResponseErrorConflict, UserDataProcessingStatusEnum>
+    >(right(UserDataProcessingStatusEnum.PENDING), retrieved => {
+      return retrieved.status === UserDataProcessingStatusEnum.CLOSED
+        ? right(UserDataProcessingStatusEnum.PENDING)
+        : left(
+            ResponseErrorConflict(
+              "An other request is already PENDING or WIP for this User"
+            )
+          );
+    });
 
-      context.log.error(`${logPrefix}|ERROR=${body}`);
+    return errorOrComputedStatus.bimap(
+      conflicError => conflicError,
+      async computedStatus => {
+        const userDataProcessing: UserDataProcessing = {
+          choice: upsertUserDataProcessingPayload.choice,
+          createdAt: new Date(),
+          fiscalCode,
+          status: computedStatus,
+          userDataProcessingId: id
+        };
+        const errorOrUpsertedUserDataProcessing = await userDataProcessingModel.createOrUpdateByNewOne(
+          userDataProcessing
+        );
+        if (isLeft(errorOrUpsertedUserDataProcessing)) {
+          const { body } = errorOrUpsertedUserDataProcessing.value;
 
-      return ResponseErrorQuery(
-        "Error while creating a new user data processing",
-        errorOrUpsertedUserDataProcessing.value
-      );
-    }
+          context.log.error(`${logPrefix}|ERROR=${body}`);
+
+          return ResponseErrorQuery(
+            "Error while creating a new user data processing",
+            errorOrUpsertedUserDataProcessing.value
+          );
+        }
 
-    const createdOrUpdatedUserDataProcessing =
-      errorOrUpsertedUserDataProcessing.value;
+        const createdOrUpdatedUserDataProcessing =
+          errorOrUpsertedUserDataProcessing.value;
 
-    const upsertedUserDataProcessingOrchestratorInput = OrchestratorInput.encode(
-      {
-        choice: createdOrUpdatedUserDataProcessing.choice,
-        fiscalCode
+        const upsertedUserDataProcessingOrchestratorInput = OrchestratorInput.encode(
+          {
+            choice: createdOrUpdatedUserDataProcessing.choice,
+            fiscalCode
+          }
+        );
+        await df
+          .getClient(context)
+          .startNew(
+            "UpsertedUserDataProcessingOrchestrator",
+            undefined,
+            upsertedUserDataProcessingOrchestratorInput
+          );
+        return ResponseSuccessJson(
+          toUserDataProcessingApi(createdOrUpdatedUserDataProcessing)
+        );
       }
-    );
-    await df
-      .getClient(context)
-      .startNew(
-        "UpsertedUserDataProcessingOrchestrator",
-        undefined,
-        upsertedUserDataProcessingOrchestratorInput
-      );
-    return ResponseSuccessJson(
-      toUserDataProcessingApi(createdOrUpdatedUserDataProcessing)
-    );
+    ).value;
```

Comment:
since you're returning the "unwrapped" value it's better to call `fold` instead of `map` / `bimap`